### PR TITLE
Add tests for IdentityServer model and queries

### LINKED ISSUE #20547: Add EF tests that use Identity Server
As a canary for breaking changes.

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #20547

### CODE REVIEW

File: test/EFCore.OData.FunctionalTests/EFCore.OData.FunctionalTests.csproj [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\EFCore.SqlServer.FunctionalTests\EFCore.SqlServer.FunctionalTests.csproj" />
+    <PackageReference Include="IdentityServer4.EntityFramework" Version="4.1.1" />
```
Comment (ajcvickers) [inline test/EFCore.OData.FunctionalTests/EFCore.OData.FunctionalTests.csproj:16]: See #23739

File: test/EFCore.AspNet.Specification.Tests/ConfigurationDbContextTestBase.cs [hunk 2]
```diff
@@ -0,0 +1,771 @@
...
+            public override DbContextOptionsBuilder AddOptions(DbContextOptionsBuilder builder)
+                => base.AddOptions(builder)
+                    .EnableDetailedErrors()
+                    .EnableSensitiveDataLogging()
+                    .ConfigureWarnings(
+                        b => b.Default(WarningBehavior.Throw)
+                            .Log(CoreEventId.SensitiveDataLoggingEnabledWarning)
+                            .Log(CoreEventId.PossibleUnintendedReferenceComparisonWarning));
+
+            protected override bool UsePooling
```
Comment (smitpatel) [inline test/EFCore.AspNet.Specification.Tests/ConfigurationDbContextTestBase.cs:767]: Any specific reason? Should add a comment here.

File: test/EFCore.AspNet.Specification.Tests/EntityTypeMapping.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+using System.Text;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Metadata;
+using Xunit;
+
+namespace Microsoft.EntityFrameworkCore
+{
+    public class EntityTypeMapping
+    {
+        public EntityTypeMapping()
```
Comment (smitpatel) [inline test/EFCore.AspNet.Specification.Tests/EntityTypeMapping.cs:15]: Is this ctor needed?

File: test/EFCore.AspNet.Specification.Tests/PersistedGrantDbContextTestBase.cs [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+            public override DbContextOptionsBuilder AddOptions(DbContextOptionsBuilder builder)
+                => base.AddOptions(builder)
+                    .EnableDetailedErrors()
+                    .EnableSensitiveDataLogging()
+                    .ConfigureWarnings(
+                        b => b.Default(WarningBehavior.Throw)
+                            .Log(CoreEventId.SensitiveDataLoggingEnabledWarning)
+                            .Log(CoreEventId.PossibleUnintendedReferenceComparisonWarning));
+
+            protected override bool UsePooling
```
Comment (smitpatel) [inline test/EFCore.AspNet.Specification.Tests/PersistedGrantDbContextTestBase.cs:364]: same

**Review by smitpatel — APPROVED** [review]

File: test/EFCore.AspNet.Specification.Tests/ConfigurationDbContextTestBase.cs (hunk 2 above)
Comment (ajcvickers) [reply test/EFCore.AspNet.Specification.Tests/ConfigurationDbContextTestBase.cs:767]: The constructors for the IdentityServer contexts have additional injected services, which means we reject them for pooling.

File: test/EFCore.AspNet.Specification.Tests/EntityTypeMapping.cs (hunk 3 above)
Comment (ajcvickers) [reply test/EFCore.AspNet.Specification.Tests/EntityTypeMapping.cs:15]: Yes. It's used when creating the expected mappings.

File: test/EFCore.AspNet.Specification.Tests/ConfigurationDbContextTestBase.cs (hunk 2 above)
Comment (smitpatel) [reply test/EFCore.AspNet.Specification.Tests/ConfigurationDbContextTestBase.cs:767]: If you can add a comment here for that 😉

**Pull request merged**